consul/autopilot.go

```diff
@@ -19,70 +20,38 @@ type AutopilotPolicy interface {
 
 func (s *Server) startAutopilot() {
 	s.autopilotShutdownCh = make(chan struct{})
+	s.autopilotWaitGroup = sync.WaitGroup{}
+	s.autopilotWaitGroup.Add(1)
 
-	go s.serverHealthLoop()
-	go s.removeDeadLoop()
+	go s.autopilotLoop()
 }
 
 func (s *Server) stopAutopilot() {
 	close(s.autopilotShutdownCh)
+	s.autopilotWaitGroup.Wait()
 }
 
-// serverHealthLoop monitors the health of the servers in the cluster
-func (s *Server) serverHealthLoop() {
+// autopilotLoop periodically looks for nonvoting servers to promote and dead servers to remove.
+func (s *Server) autopilotLoop() {
 	// Monitor server health until shutdown
-	ticker := time.NewTicker(s.config.ServerHealthInterval)
+	ticker := time.NewTicker(s.config.AutopilotInterval)
 	for {
 		select {
 		case <-s.autopilotShutdownCh:
 			ticker.Stop()
+			s.autopilotWaitGroup.Done()
```

Comment:
This is cleaner as a `defer` at the top for maintenance, in case we ever add another exit point.